### DISCUSSION TITLE
Using custom web components with van.tags

### DISCUSSION THREAD

**Discussion by deanebarker** [vanjs-org/van#234] [General]
How can I integrate VanJS with custom web components? By definition, these tags have a dash in the title, and I don't believe this works for JavaScript variable names.

So this does not work --

```
const my-web-component = van.tags
````

**Comment by Tao-VanJS** [comment]
Try this?

```js
const {"my-web-component": myWebComponent} = van.tags
```

**Comment by deanebarker** [comment]
Yes, worked perfectly. Thank you.

**Comment by deanebarker** [comment]
How about setting attributes with hyphens in the name?

```
slInput(
{
  help-text: "This clearly doesn't work..."
})
```

**Reply by Tao-VanJS** [reply]
Try this:

```js
slInput({
  "help-text": "This clearly doesn't work..."
})
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals for more information about object literals in JavaScript.

**Reply by sirenkovladd** [reply]
```js
slInput({
  "help-text": "This clearly doesn't work...",
  ["help-text"]: "This clearly doesn't work...",
  [anyVar]: "This clearly doesn't work...",
})
```

you can read more there
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#syntax
